brave/brave-core
# Add support for DNSLink resolution

Added an Open in IPFS option for domains that have a TXT record that is prefixed with dnslink=


Resolves https://github.com/brave/brave-browser/issues/13609
Resolves https://github.com/brave/brave-browser/issues/13611

## Submitter Checklist:

- [x] Requested security review https://github.com/brave/security/issues/332

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
- Navigate to brantly.eth.link and en.wikipedia-on-ipfs.org, on both you should see Open in IPFS option in address bar only if IPFS mode is local or gateway
- Click by button will open ipns://[domain] url in new tab, if you have any path in url, it should be added as well
- This should work for all domains that have dnslink= in the TXT dns record and  x-ipfs-path HTTP header is present
- This should work if page returned code different from OK 200 or failed due to network/DNS
- This must not work in private windows or in Tor windows
- Please note it doesnt work for me of I use Secure DNS "with your current service provider" as there are no available dns serviers. It may be same on your side
